runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -1,12 +1,196 @@
+        Optional<Timer.Sample> subscribeTimer = MicroMeterUtils.startTimer();
+
+        Table<LogReplicationSession, ReplicationStatus, Message> replicationStatusTable =
+            openReplicationStatusTable(corfuStore);
+
+        try {
+            return IRetry.build(IntervalRetry.class, () -> {
+                try (TxnContext txnContext = corfuStore.txn(namespace)) {
+                    // The transaction is started in the client's namespace and the Replication Status table resides in the
+                    // system namespace.  Corfu Store does not validate the cross-namespace access as long as there are no
+                    // writes on the table in the different namespace.  This hack is required here as we want client full
+                    // sync to happen in the same transaction which checks the status of a snapshot sync so that there is no
+                    // window between the check and full sync.
+                    List<CorfuStoreEntry<LogReplicationSession, ReplicationStatus, Message>> entries =
+                        txnContext.executeQuery(replicationStatusTable,
+                                record -> record.getKey().getSubscriber().getModel()
+                                    .equals(LogReplication.ReplicationModel.LOGICAL_GROUPS));
+
+                    CorfuStoreEntry<LogReplicationSession, ReplicationStatus, Message> entry = null;
+
+                    // It is possible that there is no entry for the Logical Group Model in the status table in
+                    // the following scenarios:
+                    // 1. Request to subscribe is received before the Log Replication JVM or pod starts and
+                    // initializes the table
+                    // 2. Topology change which removes this cluster from the status table.
+                    // We cannot differentiate between the two cases here so continue with the right
+                    // behavior for the 1st case, i.e., subscribe and wait for the initial Snapshot Sync to get
+                    // triggered.  If we are here because of the second case, the listener will not get any updates
+                    // from LR and subscription will be a no-op.
+                    if (entries.isEmpty()) {
+                        log.info("No record for the Logical Group Model found in the Status Table.  Subscription " +
+                                "could have been attempted before LR startup.  Subscribe the listener and wait for" +
+                                "initial Snapshot Sync");
+                    } else {
+                        // Sharding in Log Replication is done based on a destination.  So for a given replication model,
+                        // any Sink node will have a session with only 1 Source node.  Hence, there should be only 1
+                        // record corresponding to the Logical Group Replication Model.
+                        Preconditions.checkState(entries.size() == 1);
```

Comment:
If there is another client for LG this is not true, right?

Reply:
Also, this comment on sharding is not accurate. Regardless of sharding or not, if there were just one client for LG there would be one record, let sharding exist or not, right? 

Now, with sharding in place, it only holds depending on the sharding policy.

Reply:
I think without sharding there should be just 1 record? since the initial filtering already checked the client name)

Reply:
exactly @xcchang my comment was more like, this is true even without sharding so to sure the comment is very accurate. I would vote for removing the comment.